src/assets/TypeScriptAsset.js

```diff
@@ -0,0 +1,35 @@
+const JSAsset = require('./JSAsset');
+const config = require('../utils/config');
+const localRequire = require('../utils/localRequire');
+
+class TypeScriptAsset extends JSAsset {
+  async transform() {
+    super.transform();
+    
+    await this.parseIfNeeded();
+    this.isAstDirty = true;
+  }
+
+  async parse(code) {
+    // require typescript, installed locally in the app
+    let typescript = localRequire('typescript', this.name);
+
+    let transpilerOptions = {
+      compilerOptions: {
+        module: typescript.ModuleKind.CommonJS,
+        jsx: true
+      },
+      fileName: this.basename
```

Comment:
Maybe I'm missing something, but why do we need `fileName` here?

Reply:
If i'm reading the TypeScript transpile API docs correctly it should autodetect wether something is ts or tsx based on filename, this is why i included it.
``Usually TypeScript compiler uses file extension to determine if file should be parsed as '.tsx' or '.ts'. The same rule is applied during single file transpilation if the file name is provided.``

Reply:
According to the docs, `fileName` is an optional parameter of the `ts#transpile` and not a property of the tsconfig. If you take a look at the Json schema you won't find a fileName property there: http://json.schemastore.org/tsconfig

Reply:
The interface TranspileOptions which this is, does contain a FileName as far as i can see in their source code
``export interface TranspileOptions {
    compilerOptions?: CompilerOptions;
    fileName?: string;
    reportDiagnostics?: boolean;
    moduleName?: string;
    renamedDependencies?: Map<string>;
}

export function transpileModule(input: string, transpileOptions: TranspileOptions): TranspileOutput``

Reply:
Source code doesn't lie